serverLib.sml

```diff
@@ -499,18 +502,29 @@ structure ReadJSON = struct
       ,("committedDate", transform with_date read_date)
       ] empty_obj
 
+  val read_label : string option basic_reader = read_dict
+    [("name", transform (fn x => fn _ => SOME x) read_string)] NONE
```

Comment:
I'm a little wary of these lambdas in what is admittedly already (too?) complex code. Can we pull the anonymous function out and give it a name?  Maybe `transform keep_some read_string`?

Reply:
Sure. Really what's missing here is a mapM for readers. That could be used to add the SOME, otherwise it's just replace_acc. Moreover, with a mapM, read_opt_list could be split into read_list and a mapM (List.mapPartial I), and then I could just use read_list rather than figuring out how to work in a SOME.

I didn't want to make any big changes, especially since I couldn't really test anything.

Reply:
Ahem, excuse me. I mean liftM, not mapM, something which adds a function 'a -> 'b after an 'a (basic)reader to make a 'b (basic)reader.